asynction/server.py

```diff
@@ -176,35 +178,68 @@ def from_spec(
         return asio
 
     def _register_namespace_handlers(
         self,
         namespace: str,
-        channel_handlers: ChannelHandlers,
+        channel_handlers: Optional[ChannelHandlers],
         channel_bindings: Optional[ChannelBindings],
+        server_security: Optional[Sequence[SecurityRequirement]] = None,
     ) -> None:
-        if channel_handlers.connect is not None:
-            handler = load_handler(channel_handlers.connect)
+
+        on_connect = None
+
+        # if a connection handler is defined then load it
+        if channel_handlers and channel_handlers.connect is not None:
+            on_connect = load_handler(channel_handlers.connect)
 
             if self.validation:
                 with_bindings_validation = bindings_validator_factory(channel_bindings)
-                handler = with_bindings_validation(handler)
+                on_connect = with_bindings_validation(on_connect)
+
+        if server_security:
+            # if a connection handler was not defined and sever_security is present
+            # then we will inject a default handler to apply security
+            if not on_connect:
 
-            self.on_event("connect", handler, namespace)
+                def _on_connect(*args, **kwargs):
+                    return None
 
-        if channel_handlers.disconnect is not None:
-            handler = load_handler(channel_handlers.disconnect)
-            self.on_event("disconnect", handler, namespace)
+                on_connect = _on_connect
 
-        if channel_handlers.error is not None:
-            handler = load_handler(channel_handlers.error)
-            self.on_error(namespace)(handler)
+            if not self.spec.components.security_schemes:
+                raise UnregisteredSecurityScheme
+
+            # create a security handler wrapper
+            with_security = security_handler_factory(
+                server_security, self.spec.components.security_schemes
+            )
+            # apply security
+            on_connect = with_security(on_connect)
+
+        # if no user defined connection handler was specified
+        # or no security scheme was required then on_connect should still be None
+        if on_connect is not None:
+            self.on_event("connect", on_connect, namespace)
+
+        if channel_handlers:
+            if channel_handlers.disconnect is not None:
+                handler = load_handler(channel_handlers.disconnect)
+                self.on_event("disconnect", handler, namespace)
+
+            if channel_handlers.error is not None:
+                handler = load_handler(channel_handlers.error)
+                self.on_error(namespace)(handler)
 
     def _register_handlers(
-        self, default_error_handler: Optional[ErrorHandler] = None
+        self,
+        default_error_handler: Optional[ErrorHandler] = None,
+        server_security: Optional[Sequence[SecurityRequirement]] = None,
```

Comment:
```suggestion
        server_security: Sequence[SecurityRequirement],
```
based on the comment in `types.py`